android/src/main/java/com/tailscale/ipn/ui/viewModel/MainViewModel.kt

```diff
@@ -152,6 +156,16 @@ class MainViewModel(private val vpnViewModel: VpnViewModel) : IpnViewModel() {
     viewModelScope.launch {
       App.get().healthNotifier?.currentIcon?.collect { icon -> healthIcon.set(icon) }
     }
+
+    Client(viewModelScope).tailnetLockStatus { result ->
```

Comment:
This code appears to be called whenever IpnViewModel is initialized. This will detect when the device is locked out upon launch. What happens once you sign the device and the app is already running? What will trigger the locked-out state to be removed upon signing?

On iOS, we have a timer running in the background. If you get locked-out, we start fetching the lock state from LocalAPI every 3 seconds. If you are not locked out anymore, we clear the locked-out state from the UI. Ugly, but there is no other way with the current API. See `HomeViewModel.swift`.

Reply:
Closing this to instead send out a Notify when the locked out status changes.